clang/bindings/python/clang/cindex.py

```diff
@@ -2454,7 +2480,7 @@ def element_type(self):
         If accessed on a type that is not an array, complex, or vector type, an
         exception will be raised.
         """
-        result = conf.lib.clang_getElementType(self)
+        result = Type.from_result(conf.lib.clang_getElementType(self), (self,))
```

Comment:
I wonder if we should simplify this further. E.g. instead of passing the result of the library function call on `self`, and then `self` separately, should we pass just the callable and `self`, and then have the `from_result` function call the library function itself?
Would change e.g. the call marked here to `Type.from_result(conf.lib.clang_getElementType, self)`

Reply:
We definitely has more leeway here now that `from_result` function doesn't need to satisfy `errcheck` interface. And I find repeated `self` confusing. But I suggest leaving this for a later PR. Let's do this one step at a time.

Reply:
I took another look at this particular thing, even though it's out scope of this PR. Apparently, the sole purpose of this dance is to make sure that `TranslationUnit` doesn't die before `Type` or `Cursor`. I have a feeling we can do better (read "less confusing"), but I don't have a concrete proposal yet.